### PR TITLE
Bluetooth: Audio: Rename bt_codec to bt_audio_codec_{cap, conf, data}

### PR DESCRIPTION
**Pull request by Thalley**
Rename the bt_codec struct to bt_audio_codec_conf or to the new struct bt_audio_codec_cap.

The purpose of this is to split the codec specific configuration and codec capabilities into seperate structs, as they do not reflect the same values, or used for the same purpose.

This commit depends on the preset macros workings on either type of struct (for now), but will be modified in future updates.

First step for fixing https://github.com/zephyrproject-rtos/zephyr/issues/42860

### CODE REVIEW

**Comment by Thalley** [comment]
Comment for reviewers: Suggest to focus review on the public header file changes.

**Review by kruithofa — DISMISSED** [review]

**Review by MariuszSkamra — DISMISSED** [review]

**Comment by Thalley** [comment]
Marked as DNM as I rather postpone this rename until after the next release so that it can be added in the same release as the rest of the changes for https://github.com/zephyrproject-rtos/zephyr/issues/42860

**Comment by Thalley** [comment]
Rebased to fix some CI issues

**Comment by Thalley** [comment]
Rebased to fix some CI issues

**Review by MariuszSkamra — DISMISSED** [review]

**Comment by Thalley** [comment]
Rebased to solve merge conflicts

**Comment by Thalley** [comment]
@sjanc This PR also modifies some code in the BTTester. Are we building that during CI? Just want to make sure I don't stuff.

**Review by kruithofa — DISMISSED** [review]

**Comment by Thalley** [comment]
Rebased to solve merge conflicts

**Review by larsgk — APPROVED** [review]

**Review by larsgk — DISMISSED** [review]

File: include/zephyr/bluetooth/audio/audio.h [hunk 1]
```diff
@@ -430,21 +456,16 @@ struct bt_codec_qos {
...
-		.rtn = _rtn, \
-		.latency = _latency, \
-		.pd_min = _pd_min, \
-		.pd_max = _pd_max, \
-		.pref_pd_min = _pref_pd_min, \
-		.pref_pd_max = _pref_pd_max, \
+#define BT_AUDIO_CODEC_QOS_PREF(_unframed_supported, _phy, _rtn, _latency, _pd_min, _pd_max,       \
+				_pref_pd_min, _pref_pd_max)                                        \
+	{                                                                                          \
+		.unframed_supported = _unframed_supported, .phy = _phy, .rtn = _rtn,               \
```
[the remark this answers is not in the document]
Comment (Thalley) [reply include/zephyr/bluetooth/audio/audio.h]: Fixed

**Comment by Thalley** [comment]
Rebased to solve merge conflicts

**Review by cvinayak — APPROVED** [review]
quick-ack; PR appears to be approved by multiple reviewers and is refactoring/renaming, LGTM.

**Review by larsgk — APPROVED** [review]

**Review by kruithofa — APPROVED** [review]

**Pull request merged**